tests/integration/multicluster/remotekubeconfig/main_test.go

```diff
@@ -0,0 +1,130 @@
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/tests/integration/multicluster"
+
+	kubeApiCore "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	ist istio.Instance
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		Label(label.Multicluster).
+		RequireMinClusters(2).
+		Setup(kube.Setup(func(s *kube.Settings, ctx resource.Context) {
+			// Make istiod run on 2nd cluster and points to KUBECONFIG in cluster 1
+			scopes.Framework.Infof("setting up istiod cluster before hand")
+			s.ControlPlaneTopology = make(map[resource.ClusterIndex]resource.ClusterIndex)
+			primaryCluster := resource.ClusterIndex(0)
+			cfg, err := istio.DefaultConfig(ctx)
+			if err != nil {
+				scopes.Framework.Infof("has error in creating istio cfg ")
+				return
+			}
+			for i := 0; i < len(s.KubeConfig); i++ {
+				s.ControlPlaneTopology[resource.ClusterIndex(i)] = primaryCluster
+			}
+			//create related namespace ,secret and service account before hand in 2nd cluster
+			istioKubeConfig, err := file.AsString(s.KubeConfig[0])
+			if err != nil {
+				scopes.Framework.Infof("has error in parsing kubeconfig ")
+				return
+			}
+			istiodNS := &kubeApiCore.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: cfg.SystemNamespace,
+				},
+			}
+			ctx.Clusters()[1].CoreV1().Namespaces().Create(context.TODO(),
+				istiodNS, metav1.CreateOptions{})
+
+			istiokubeconfigSecret := &kubeApiCore.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "istio-kubeconfig",
+				},
+				Data: map[string][]byte{
+					"config": []byte(istioKubeConfig),
+				},
+			}
+			_, err = ctx.Clusters()[1].CoreV1().Secrets(cfg.SystemNamespace).Create(context.TODO(),
+				istiokubeconfigSecret, metav1.CreateOptions{})
+			if err != nil {
+				scopes.Framework.Infof("has error in creating istio-kubeconfig secrets %v", err)
+				return
+			}
+			istiodSA := &kubeApiCore.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: cfg.SystemNamespace,
+					Name:      "istiod-service-account",
+				},
+			}
+			_, err = ctx.Clusters()[1].CoreV1().ServiceAccounts(cfg.SystemNamespace).Create(context.TODO(),
+				istiodSA, metav1.CreateOptions{})
+			if err != nil {
+				scopes.Framework.Infof("has error in creating istiod service account %v", err)
+				return
+			}
+		})).
+		Setup(istio.Setup(&ist, func(cfg *istio.Config) {
+			// Set the control plane values on the config.
+			cfg.UsingRemoteKubeConfig = true
+			cfg.ControlPlaneValues =
+				`components:
+  base:
+    enabled: true
+  pilot:
+    enabled: false
+  telemetry:
+    enabled: false
+  istiodRemote:
+    enabled: true
+  ingressGateways:
+  - enabled: false`
+			cfg.RemoteClusterValues = `
+components:
+  base:
+    enabled: false
+  pilot:
+    enabled: true
+  ingressGateways:
+  - enabled: false
+values:
+  global:
+    operatorManageWebhooks: true
+`
+		})).
+		Run()
+}
+
+func TestIngressGateWay(t *testing.T) {
```

Comment:
Can you add some comments about what this is testing?

Reply:
The whole flow is:
1. create istio-kubeconfig secret in control cluster
2. install istiod remote + base in remote cluster
3. install istiod in control cluster
4. patch istiod service as LoadBalancer and customHost in control cluster
4. patch istio cm and two webhook configurations in remote cluster

Then this test is to install a istio ingress gateway in remote cluster and make sure it can be started successfully without  issue.

Reply:
I mean in the code :slightly_smiling_face: